packages/desktop-client/src/components/reports/reports/Custom.js

```diff
@@ -0,0 +1,344 @@
+import Text from '../../common/Text';
+import View from '../../common/View';
+import { AppliedFilters } from '../../filters/FiltersMenu';
+import PrivacyFilter from '../../PrivacyFilter';
+import { ChooseGraph } from '../ChooseGraph';
+import Header from '../Header';
+import { ReportOptions } from '../ReportOptions';
+import { CustomSidebar } from '../ReportSidebar';
+import { ReportLegend, ReportSummary } from '../ReportSummary';
+import { CustomTopbar } from '../ReportTopbar';
+import defaultSpreadsheet from '../spreadsheets/default-spreadsheet';
+import useReport from '../useReport';
+import { fromDateRepr } from '../util';
+
+export default function Custom() {
+  const categories = useCategories();
+
+  let { payees, accounts } = useSelector(state => {
+    return {
+      payees: state.queries.payees,
+      accounts: state.queries.accounts,
+    };
+  });
+
+  const {
+    filters,
+    conditionsOp,
+    onApply: onApplyFilter,
+    onDelete: onDeleteFilter,
+    onUpdate: onUpdateFilter,
+    onCondOpChange,
+  } = useFilters();
+
+  const [selectedCategories, setSelectedCategories] = useState(null);
+  const [allMonths, setAllMonths] = useState(null);
+  const [typeDisabled, setTypeDisabled] = useState(['Net']);
+  const [start, setStart] = useState(
+    monthUtils.subMonths(monthUtils.currentMonth(), 5),
+  );
+  const [end, setEnd] = useState(monthUtils.currentMonth());
+
+  const [mode, setMode] = useState('total');
+  const [groupBy, setGroupBy] = useState('Category');
+  const [balanceType, setBalanceType] = useState('Expense');
+  const [empty, setEmpty] = useState(false);
+  const [hidden, setHidden] = useState(false);
+  const [uncat, setUncat] = useState(false);
+  const [dateRange, setDateRange] = useState('6 months');
+
+  const [graphType, setGraphType] = useState('BarGraph');
+  const [viewLegend, setViewLegend] = useState(false);
+  const [viewSummary, setViewSummary] = useState(false);
+  const [viewLabels, setViewLabels] = useState(false);
+  //const [legend, setLegend] = useState([]);
+  let legend = [];
+  const dateRangeLine = ReportOptions.dateRange.length - 1;
+
+  const months = monthUtils.rangeInclusive(start, end);
+  const getGraphData = useMemo(() => {
+    return defaultSpreadsheet(
+      start,
+      end,
+      groupBy,
+      ReportOptions.balanceType.find(opt => opt.description === balanceType)
+        .format,
+      categories,
+      selectedCategories,
+      payees,
+      accounts,
+      filters,
+      conditionsOp,
+      hidden,
+      uncat,
+    );
+  }, [
+    start,
+    end,
+    groupBy,
+    balanceType,
+    categories,
+    selectedCategories,
+    payees,
+    accounts,
+    filters,
+    conditionsOp,
+    hidden,
+    uncat,
+  ]);
+  const data = useReport('default', getGraphData);
+
+  useEffect(() => {
+    if (selectedCategories === null && categories.list.length !== 0) {
+      setSelectedCategories(categories.list);
+    }
+  }, [categories, selectedCategories, groupBy]);
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏don't think `groupBy` is necessary here
```suggestion
  }, [categories, selectedCategories]);
```